libavcodec/cbs_h266_syntax_template.c

```diff
@@ -3587,10 +3588,127 @@ SEI_FUNC(sei_buffering_period, (CodedBitstreamContext *ctx, RWContext *rw,
+            else
+                infer(pt_cpb_removal_delay_delta_enabled_flag[i], 0);
+
+            if (current->pt_cpb_removal_delay_delta_enabled_flag[i])
+                if (bp->bp_num_cpb_removal_delay_deltas_minus1 > 0)
+                    ubs(av_ceil_log2(bp->bp_num_cpb_removal_delay_deltas_minus1 + 1), pt_cpb_removal_delay_delta_idx[i], 1, i);
+                else
+                    infer(pt_cpb_removal_delay_delta_idx[i], 0);
+            else
+                ubs(cpb_removal_delay_length, pt_cpb_removal_delay_minus1[i], 1, i);
+        }
+    }
+
+    cpb_initial_removal_delay_length = bp->bp_cpb_initial_removal_delay_length_minus1 + 1;
+    dpb_output_delay_length = bp->bp_dpb_output_delay_length_minus1 + 1;
+    ub(dpb_output_delay_length, pt_dpb_output_delay);
+
+    if (bp->bp_alt_cpb_params_present_flag) {
+        flag(pt_cpb_alt_timing_info_present_flag);
+        if (current->pt_cpb_alt_timing_info_present_flag) {
+            if (bp->bp_nal_hrd_params_present_flag) {
+                for (i = (bp->bp_sublayer_initial_cpb_removal_delay_present_flag ? 0 :
+                     bp->bp_max_sublayers_minus1); i <= bp->bp_max_sublayers_minus1; i++) {
+                    for (j = 0; j < bp->bp_cpb_cnt_minus1 + 1; j++) {
+                        ubs(cpb_initial_removal_delay_length, pt_nal_cpb_alt_initial_removal_delay_delta[i][j],  2, i, j);
+                        ubs(cpb_initial_removal_delay_length, pt_nal_cpb_alt_initial_removal_offset_delta[i][j], 2, i, j);
+                    }
+                    ubs(cpb_removal_delay_length, pt_nal_cpb_delay_offset[i][j], 2, i, j);
+                    ubs(cpb_removal_delay_length, pt_nal_dpb_delay_offset[i][j], 2, i, j);
+                }
+            }
+            if (bp->bp_vcl_hrd_params_present_flag) {
+                for (i = (bp->bp_sublayer_initial_cpb_removal_delay_present_flag ? 0 :
+                     bp->bp_max_sublayers_minus1); i <= bp->bp_max_sublayers_minus1; i++) {
+                    for (j = 0; j < bp->bp_cpb_cnt_minus1 + 1; j++) {
+                        ubs(cpb_initial_removal_delay_length, pt_vcl_cpb_alt_initial_removal_delay_delta[i][j],  2, i, j);
+                        ubs(cpb_initial_removal_delay_length, pt_vcl_cpb_alt_initial_removal_offset_delta[i][j], 2, i, j);
+                    }
+                    ubs(cpb_removal_delay_length, pt_vcl_cpb_delay_offset[i][j], 2, i, j);
+                    ubs(cpb_removal_delay_length, pt_vcl_dpb_delay_offset[i][j], 2, i, j);
+                }
+            }
+        }
+    } else {
+        infer(pt_cpb_alt_timing_info_present_flag, 0);
+    }
+
+    if (bp->bp_du_hrd_params_present_flag && bp->bp_du_dpb_params_in_pic_timing_sei_flag)
+        ub(bp->bp_dpb_output_delay_du_length_minus1 + 1, pt_dpb_output_du_delay);
+
+    if (bp->bp_du_hrd_params_present_flag && bp->bp_du_cpb_params_in_pic_timing_sei_flag) {
+        ue(pt_num_decoding_units_minus1, 0, 0xffffffff);
```

Comment:
The value of pt_num_decoding_units_minus1 shall be in the range of 0 to PicSizeInCtbsY − 1, inclusive.